src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java

```diff
@@ -374,9 +375,8 @@ private HttpResponse sendAndDefer(HttpRequest request, HttpRequestOptions option
   /**
    * Blocking send which waits to load the whole response to memory before propagating it.
    */
-  private HttpResponse sendAndWait(HttpRequest request, HttpRequestOptions options, int currentRedirects)
+  private HttpResponse sendAndWait(HttpRequest request, Request grizzlyRequest, HttpRequestOptions options, int currentRedirects)
```

Comment:
Why is the need for a request and grizzlyRequest. Add documentation indicating why this is needed.

Reply:
Because I made this method recursive. In case the server response is a 30x we create a new request in line 357 and call this method again.

Reply:
I think the problem is that you don't have a way to set the cookies and the remove the headers in the same object also, right?